core/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -35,52 +39,87 @@
 import java.util.List;
 
 /**
- * Information about snapshot
+ * Information about a snapshot
  */
-public class SnapshotInfo implements ToXContent, Streamable {
+public final class SnapshotInfo implements Comparable<SnapshotInfo>, ToXContent, FromXContentBuilder<SnapshotInfo>, Writeable {
```

Comment:
I'm really not a fan of the `FromXContentBuilder` interface. I figure for the most part we shouldn't be implementing it - just making static method references to static `fromXContent` methods?